sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/sqs/SqsIO.java

```diff
@@ -709,8 +789,9 @@ private class Dynamic extends Batches {
             (queue, batch) -> batch != null && batch.lock(true) ? batch : createLocked(queue);
 
         private final Map<@NonNull String, Batch> batches = new HashMap<>();
+        private final AtomicBoolean submitExpiredRunning = new AtomicBoolean(false);
```

Comment:
Why do we need atomic types here if it's not thread safe?

Reply:
The class is not entirely thread-safe, but certain operations have to be, e.g. `submitExpired(boolean throwFailures)`